includes/transcripts.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Adds an endpoint for viewing transcripts.
+ *
+ * @package tenup_podcasting\transcripts;
+ */
+
+namespace tenup_podcasting\transcripts;
+
+use DOMDocument;
+use WP_Post;
+
+/**
+ * Wrap unwrapped text in a paragraph tag.
+ *
+ * @param string $text The incoming markup.
+ *
+ * @return string
+ */
+function podcasting_wrap_unwrapped_text_in_paragraph( $text ) {
+	$doc = new DOMDocument();
+	libxml_use_internal_errors( true );
+	$doc->loadHTML( '<html><body>' . $text . '</body></html>' );
+	$body_node     = $doc->getElementsByTagName( 'body' )->item( 0 );
+	$filtered_text = '';
+
+	// phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+	foreach ( $body_node->childNodes as $node ) {
+		// phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+		if ( XML_TEXT_NODE === $node->nodeType ) {
+			$filtered_text .= '<p>' . $doc->saveHTML( $node ) . '</p>';
```

Comment:
I'm seeing some strange markup here when using the citation button.

The markup is rendered as

```html
<cite>Person 1</cite>
<p>: Welcome to Show</p>
<br>
<cite>Person 2</cite>
<p>: I'm P2</p>
<br>
<cite>Person 1:</cite>
<p> and I'm P1.</p>
<br>
</body>
```

The markup is saved in post meta as:

```html
<cite>Person 1</cite>: Welcome to Show<br>
<cite>Person 2</cite>: I'm P2<br>
<cite>Person 1:</cite> and I'm P1.<br>
```

Reply:
I was trying to make RichText do too much. This should have been InnerBlocks, and has been changed. The added benefit is that this is much more user friendly now.